### ISSUE TITLE
Bare ClassVar should infer type instead of defaulting to Any

### ISSUE THREAD

**Issue by ilevkivskyi** [python/mypy#5587]
Currently we have:
```python
from typing import ClassVar
  
class C:
    x: ClassVar = 1

reveal_type(C.x)  # Revealed type is 'Any'
```
`ClassVar` is not a generic type (and not really a type, rather an access qualifier). So it will be better if a bare `ClassVar` would use inference instead of defaulting to `Any` thus reducing verbosity or false negatives (although `--disallow-any-generics` actually flags this). 

Probably not high priority.

**Comment by gvanrossum** [comment]
I agree. Should this become an update to PEP 526?

**Comment by ilevkivskyi** [comment]
> Should this become an update to PEP 526?

I think it is worth explicitly mentioning it in the PEP.

**Comment by gvanrossum** [comment]
OK, but is it still provisional? It was provisionally accepted for 3.6. Can we still change it? (PEP 411 suggests provisional applies to a single feature release.)

**Comment by gvanrossum** [comment]
Fair enough.

**Comment by Avasam** [comment]
This is now part of the typing spec:
- https://github.com/python/typing-council/issues/43
- https://typing.python.org/en/latest/spec/class-compat.html#classvar
